scripts/dialectic_cost_calculator.py

```
@@ -0,0 +1,439 @@
+        f"  Total input tokens (all iterations): {max_result['total_input_tokens']:,}"
+    )
+    console.print(
+        f"    - Uncached: {max_result['total_uncached_tokens']:,} @ ${MODEL_PRICING[max_result['model']]['input']}/1M"
+    )
+    console.print(
+        f"    - Cached: {max_result['total_cached_tokens']:,} @ ${MODEL_PRICING[max_result['model']]['cached']}/1M"
+    )
+    console.print("  Output tokens:")
+    console.print(
+        f"    - Realistic: {max_result['total_output_tokens_realistic']:,} "
+        + f"(9 tool calls × {estimates.realistic_thinking_per_tool + estimates.realistic_tool_call_output} + final {max_result['thinking_tokens'] + estimates.realistic_final_answer})"
```

Ruff (0.14.13)

String contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF001)